kbcstorage/workspaces.py

```diff
@@ -0,0 +1,112 @@
+"""
+Manages workspace requests to the API.
+
+Full documentation `here`.
+
+.. _here:
+    http://docs.keboola.apiary.io/#reference/workspaces/
+"""
+
+from kbcstorage.base import Endpoint
+
+
+def _make_body(mapping):
+    """
+    Given a dict mapping Keboola tables to aliases, construct the body of
+    the HTTP request to load said tables.
+
+    Args:
+        mapping(:obj:`dict`): Keys contain the full names of the tables to
+            be loaded (ie. 'in.c-bucker.table_name') and values contain the
+            aliases to which they will be loaded (ie. 'table_name').
+        preserve(bool): If True, does not clear the workspace of existing
+            tables. If False, clears the workspace of tables before loading.
+            Default False.
+    """
+    body = {}
+    template = 'input[{0}][{1}]'
+    for i, (k, v) in enumerate(mapping.items()):
+        body[template.format(i, 'source')] = k
+        body[template.format(i, 'destination')] = v
+
+    return body
+
+
+class Workspaces(Endpoint):
+    """
+    Workspaces Endpoint
+    """
+    def __init__(self, url, token):
+        """
```

Comment:
I'm playing with the idea that it might be better to pass the endpoint class to workspace class as a ctor parameter - it's just that this way one has to pass the token all around the code and it's bound to leak somewhere...